packages/dapp/src/helpers/parameterizer.ts

```diff
@@ -38,20 +49,72 @@ export async function initializeProposalsSubscriptions(dispatch: Dispatch<any>):
       ? await parameterizer.getPropChallengeRevealExpiry(propID)
       : undefined;
     const propProcessByExpiry = await parameterizer.getPropProcessBy(propID);
-    dispatch(
-      addOrUpdateProposal({
-        id: propID,
-        paramName,
-        propValue,
-        state: propState,
-        applicationExpiry,
-        propProcessByExpiry,
-        challenge: {
-          id: challengeID,
-          challengeCommitExpiry,
-          challengeRevealExpiry,
-        },
-      }),
-    );
+    const paramProposal = {
+      id: propID,
+      paramName,
+      propValue,
+      state: propState,
+      applicationExpiry,
+      propProcessByExpiry,
+      challenge: {
+        id: challengeID,
+        challengeCommitExpiry,
+        challengeRevealExpiry,
+      },
+    };
+    await setupParamProposalCallback(paramProposal, true, dispatch);
+    dispatch(addOrUpdateProposal(paramProposal));
   });
 }
+
+async function getNextTimerExpiry(paramProposal: any, dispatch: Dispatch<any>): Promise<number> {
+  let nextExpiry;
+
+  switch (paramProposal.state) {
+    case ParamProposalState.APPLYING:
+      nextExpiry = paramProposal.applicationExpiry.valueOf() / 1000;
```

Comment:
this is definitely correct? I would've assume it would be `* 1000` since the contract is in seconds but timeouts are in milliseconds

Reply:
or just not divided by anything

Reply:
Yeah, these are correct actually. The dapp is using the Parameterizer getters for these expiry values, which each return Dates.

https://github.com/joincivil/Civil/blob/master/packages/core/src/contracts/tcr/parameterizer.ts#L383

This could be cleaned up a bit though... let me make a few changes.